src/App/FreeCADInit.py

```diff
@@ -86,33 +101,46 @@ def InitApplications():
 	# add also this path so that all modules search for libraries
 	# they depend on first here
 	PathExtension = BinDir + os.pathsep
+
 	# prepend all module paths to Python search path
 	Log('Init:   Searching for modules...\n')
-	FreeCAD.__path__ = ModDict.values()
+
+
+	# to have all the module-paths available in FreeCADGuiInit.py:
+	FreeCAD.__ModDirs__ = list(ModDict.values())
+
+	# this allows importing with:
+	# from FreeCAD.Module import package
+	FreeCAD.__path__ = [ModDir, Lib64Dir, LibDir, HomeMod]
+
+	# also add these directories to the sys.path to 
+	# not change the old behaviour. once we have moved to 
+	# proper python modules this can eventuelly be removed.
+	sys.path = [ModDir, Lib64Dir, LibDir] + sys.path
+
 	for Dir in ModDict.values():
 		if ((Dir != '') & (Dir != 'CVS') & (Dir != '__init__.py')):
 			sys.path.insert(0,Dir)
 			PathExtension += Dir + os.pathsep
 			InstallFile = os.path.join(Dir,"Init.py")
 			if (os.path.exists(InstallFile)):
 				try:
-					#execfile(InstallFile)
-					exec open(InstallFile).read()
-				except Exception, inst:
+					# XXX: This looks scary securitywise...
+
+					with open(InstallFile) as f:
+						exec(f.read())

```

Comment:
This is not about modules. it's about the freecad Init for a package. I also thought of an alternative, but somehow it wasn't working when I tried it. Don't remember the problem.